### ISSUE TITLE
Include trades and asset movements from non-connected exchanges in history

### ISSUE THREAD

**Issue by LefterisJP** [rotki/rotki#1864]
## Problem Definition

If all trades are queried, only trades from connected exchanges are returned:

https://github.com/rotki/rotki/blob/f3a795e0a3061751db22e6ebfcf7d4701488878d/rotkehlchen/rotkehlchen.py#L603

If asset movements are queried, they are only queried from the connected exchanges or crypto.com:

https://github.com/rotki/rotki/blob/f3a795e0a3061751db22e6ebfcf7d4701488878d/rotkehlchen/rotkehlchen.py#L847-L888

That can miss:

- Trades/asset movements imported from cointracking
- Trades/asset movements from non-connected exchanges that are in the DB. They would most probably be from cointracking though since when we delete an api key connection we clean the DB.

## Task

Make it also fetch any of the non-connected location entries from the DB and include them in the result

**Comment by kelsos** [comment]
@LefterisJP the way we do the checks at the moment requires this to also be added on the frontend.
The current approach though it's not perfect since for example crypto.com is queried every single time even for users that don't have any data.

For normal exchanges, we have the connected exchanges property so only these are queried by the frontend for things like crypto.com and cointracking we don't have anything though. How easy or hard would be to notify the frontend that something exists there? to avoid having hardcoded always performed requests?

**Comment by LefterisJP** [comment]
@yabirgb don't you actually solve this already in v1.17.0?

**Comment by LefterisJP** [comment]
All right, so please make sure to also close this issue then once you are done.

**Comment by yabirgb** [comment]
For the first point `Trades/asset movements imported from cointracking`they should be correctly queried if the location is activated. 

About how to handle the trades from exchanges not connected we'll spend more time since essentially we need to ask for every known exchange, and the frontend also needs to query for all exchanges. I believe a better solution will be to add a list of imported exchanges in addition to the existing `conneted_exchanges` and use it at the moment of querying trades (this list is not the same as the external locations since will only contain locations where we have data but not from connected exchanges)

**Comment by kelsos** [comment]
Having the list of external locations with trades/asset movements would certainly be a better approach than having to maintain a list of external locations on the frontend. Also this way we won't do any unneeded queries.

**Comment by yabirgb** [comment]
So to close this issue after #3797 gets merged we need to include all the trades in the frontend queries. Currently the frontend creates different trades queries for the external locations (crypto.com, bisq, nexo...) and for the exchanges that it knows have keys.

The PR #3797 introduces a new endpoint that will return all the locations that contain information, this can be exchanges, external locations or some defi protocols like `uniswap`, `sushiswap` and `balancer`.

What the frontend should do now is instead of trusting the known list of exchanges and query all external locations call this endpoint instead and query trades only for this locations. Take into account that it can include locations like `uniswap` and other so maybe when you query the trades you will need to apply some logic in the returned list
